manifests/utility/database.pp

```diff
@@ -0,0 +1,24 @@
+# Sample Usage:
+#   mariadb::utility::database { "production dbname" :
+#       dbname => 'dbname',
+#       charset => 'utf8',
+#       collate => 'utf8_general_ci'
+#       ensure => 'present',
+#   }
+define mariadb::utility::database($dbname, $charset = 'utf8', $collate = 'utf8_general_ci', $ensure = 'present' ) {
```

Comment:
I see this as neccesairy for multiple reason;
- usuability. One would expect them here. You can pretty much setup the Server in one a oneliner, why not ensure that users dbs and grants are also included.
- One does not necessairy need to know about the core module to use it
- Furthermore there should be added validation anyway. So far i have added just ensure but ther could be more
- finally when i see a project like this, i think we agree that the main reason for doing this is getting the show on the road quickly. this makes it easy and its not like its including megabytes of unneccesairy code.

i can totally see though if you see this as clutter, in that case just replace it with core module calls =)